client/my-sites/upgrades/map-domain/index.jsx

```diff
@@ -64,11 +64,18 @@ var MapDomain = React.createClass( {
 	},
 
 	goBack: function() {
-		if ( ! this.props.sites ) {
-			return page( this.props.path.replace( '/mapping', '' ) );
+		const selectedSite = this.props.sites && this.props.sites.getSelectedSite();
+		if ( ! selectedSite ) {
+			page( this.props.path.replace( selectedSite.is_vip ? '/add/mapping' : '/mapping', '' ) );
+			return;
```

Comment:
`if ( ! selectedSite )` followed by `selectedSite.is_vip` looks like a logical error to me. I think `getSelectedSite()` returns false which is why this wont generate a JS error. Either way I feel like something unexpected happens here when a site is not selected.